src/validations/access/on-chain/client.ts

```diff
@@ -91,38 +97,50 @@ export function createOnChainClient(
     }
 
     console.log('urnsToCheck', urnsToCheck)
-    const { ethereum, matic } = await splitItemsURNsByNetwork(urnsToCheck)
-    console.log('ethereum', ethereum, 'matic', matic)
+
+    const { ethereum, matic, ethereumThirdParty, maticThirdParty } = await splitItemsURNsByTypeAndNetwork(urnsToCheck)
+    console.log('ethereumThirdParty', ethereumThirdParty, 'maticThirdParty', maticThirdParty)
 
     const ignoredSet = new Set([
       ...ethereum.filter(({ type }) => type === 'blockchain-collection-v1-asset').map(({ urn }) => urn),
-      ...matic.filter(({ type }) => type === 'blockchain-collection-v2-asset').map(({ urn }) => urn)
+      ...matic.filter(({ type }) => type === 'blockchain-collection-v2-asset').map(({ urn }) => urn),
+      // ...ethereumThirdParty should never contain assets, so no need to filter
+      ...maticThirdParty.filter(({ type }) => type === 'blockchain-collection-third-party').map(({ urn }) => urn)
     ])
     console.log('ignoredSet', ignoredSet)
     if (ignoredSet.size > 0) {
       logger.info(`Ignoring these assets, considering them as "owned" by the address: ${[...ignoredSet]}`)
     }
+
     const filteredEthereum = ethereum
-      .filter(({ type }) => type !== 'blockchain-collection-v1-asset')
+      .filter(({ type }) => type === 'blockchain-collection-v1-item')
+      .map(({ urn }) => urn)
+    const filteredMatic = matic.filter(({ type }) => type === 'blockchain-collection-v2-item').map(({ urn }) => urn)
+    const filteredEthereumThirdParty = ethereumThirdParty
+      .filter(({ type }) => type === 'blockchain-collection-third-party-item')
+      .map(({ urn }) => urn)
```

Comment:
Line 107 says: `// ...ethereumThirdParty should never contain assets, so no need to filter`.
Would you mind explaining why using the filter here and not before when creating the `ignoredSet`? 🙏